src/lualib/ArrayConcat.ts

```diff
@@ -0,0 +1,24 @@
+/* tslint:disable */
+declare function pcall(func: Function): any;
+declare function type(val: any): string;
+
+function __TS__ArrayConcat(arr1: any[], ...args: any[]): any[] {
+  const out: any[] = [];
+  for (let i = 0; i < arr1.length; i++) {
+    out[out.length] = arr1[i];
+  }
+  for (let i = 0; i < args.length; i++) {
+    const arg = args[i];
+    // Hack because we don't have an isArray function
+    if (pcall(() => (arg as any[]).length) && type(arg) !== "string") {
```

Comment:
Cant really make assumptions here a `type(arg) == "table"` won't work since you are allowed to pass objects to concat.

Only other solution I could think of is iterating the table with pairs and checking if each key is a number, I didn't choose that variant for performance reasons.